tests/conftest.py

```diff
@@ -5,6 +5,12 @@
 from cryptography.hazmat.backends import default_backend
 from cryptography.hazmat.primitives import serialization
 
+from uvicorn.config import LOGGING_CONFIG
+
+# Workaround: pytest caplog not able to capture no-propagate loggers
```

Comment:
I'd leave a more detailed explanation here as to why we explicitly turn the propagate on just for tests, mainly for the maintainers looking at that part because it's really not obvious, something from your comment here would be cool :
https://github.com/encode/uvicorn/pull/1288#issuecomment-991585560

Reply:
@euri10 resolved